lib/segment/src/index/visited_pool.rs

```diff
@@ -14,52 +14,81 @@ const POOL_KEEP_LIMIT: usize = 16;
 /// It stores the sequence number of last processed operation next to the point ID, which allows to avoid memory allocation
 /// and reuse same counter for multiple queries.
 #[derive(Debug)]
-pub struct VisitedList {
+pub struct VisitedList<'a> {
```

Comment:
What about naming this type `VisitedListHandle` and keeping the other one as `VisitedList`.

I think I'd like that better because 'handle' more clearly states that it is a pointer to external state. But I don't have a strong feeling for either of the two options. Please choose whichever you prefer.

Reply:
I like your approach. I'm bad with namings. Will change to `Handle`